bin/docker-build-cli-bin

```diff
@@ -22,22 +22,28 @@ validate_go_deps_tag $dockerfile
     $bindir/docker-build-go-deps
 ) >/dev/null
 
+CURRENT_PLATFORM=$(uname)
```

Comment:
We do need a way to build the Docker container the way it is now, with all three platforms included in the container, by executing bin/docker-build. The reason is that the official release process, which isn't documented in this open source repo yet (I hope to do so soon) consists of `bin/docker-build` followed by uploading all three platforms' `conduit` CLI apps to GitHub (amongst other steps).

Also, more generally, it might be confusing and/or error-prone to build the CLI docker image in a way that's different from its normal intended configuration. And also, if we can avoid Docker in the first place, it would be faster yet. On my laptop, as much of the slowness of building the CLI comes from Docker and virtual machine overhead as it does from the actual `go build` steps in the Dockerfile.

I suggest that we do this a bit differently:
* Create a `bin/fast-build` script
* Inside `bin/fast-build`, execute something like `CONDUIT_SKIP_CLI_CONTAINER=1 bin/docker-build`
* In `bin/docker-build`, skip `docker-build-cli-bin` completely if `$CONDUIT_SKIP_CLI_CONTAINER` is set.
* In `bin/fast-build`, add a step that does `CGO_ENABLED=0 bin/go-build  go build -installsuffix cgo -o target/cli/${OS}/conduit -ldflags "${GO_LDFLAGS}" where `GO_LDFLAGS` is set like it is in the Dockerfile.
* (Perhaps as a follow-up) Find some convenient way to centralize the command line flags for `go build` and `go install` in one file that is used by all the Dockerfiles and the fast-build script.

This way, we would create a fast-path that eliminates all the unnecessary overhead, both Docker and the waste of building for usually-irrelevant platforms, with minimal risk to the production build system.

WDYT?

Reply:
is the docker really such a bottleneck? my only problem with the thing that you are suggesting is that we will introduce a second build script that is not run in CI... so it is potentially one more thing that have to be maintained in the future and if some breaking change in a way the CLI is built is introduced, this new script might get broken and we won't even know that from CI...

I would personally stick with one way to build this but if you say that the performance gain is actually worth it, I'll implement your way :)

let me know...

Reply:
@briansmith ping? :)

Reply:
OK. I think it is fine to limit the scope of this PR to more what you originally submitted. Below I suggest smaller changes that are necessary for it to be merged.